experiment/selinux/src/selinux.rs

```diff
@@ -44,268 +95,301 @@ impl SELinux {
+                "can't open the config file: {}",
+                e
+            ))),
         }
-        Ok(())
     }
 
-    // function similar with lSetFileLabel in go-selinux repo.
-    // lset_file_label sets the SELinux label for this path, not following symlinks,
-    // or returns an error.
-    pub fn lset_file_label(fpath: &Path, label: &str) -> Result<(), SELinuxError> {
-        if !fpath.exists() {
-            return Err(SELinuxError::LSetFileLabel(ERR_EMPTY_PATH.to_string()));
+    // get_enabled returns whether SELinux is enabled or not.
+    pub fn get_enabled(&mut self) -> bool {
+        match Self::get_selinux_mountpoint(self) {
+            // If there is no SELinux mountpoint, SELinux is not enabled.
+            None => false,
+            Some(_) => match Self::current_label(self) {
+                Ok(con) => {
+                    if con != "kernel" {
+                        return true;
+                    }
+                    false
+                }
+                Err(_) => false,
+            },
         }
+    }
 
+    // verify_selinux_fs_mount verifies if the specified mount point is
+    // properly mounted as a writable SELinux filesystem.
+    fn verify_selinux_fs_mount<P: AsRef<Path>>(mnt: P) -> bool {
+        let mnt = mnt.as_ref();
         loop {
-            match fpath.lset_xattr(XATTR_NAME_SELINUX, label.as_bytes()) {
-                Ok(_) => break,
-                // When a system call is interrupted by a signal, it needs to be retried.
-                Err(XattrError::EINTR(_)) => continue,
-                Err(e) => {
-                    return Err(SELinuxError::LSetFileLabel(format!(
-                        "lset_xattr failed: {}",
-                        e
-                    )));
+            match statfs::statfs(mnt) {
+                Ok(stat) => {
+                    // verify if the file is readonly or not
+                    if !stat.flags().contains(statvfs::FsFlags::ST_RDONLY) {
+                        return false;
+                    }
+                    // verify if the file is SELinux filesystem
+                    return stat.filesystem_type() == statfs::SELINUX_MAGIC;
                 }
+                // check again if there is an issue while calling statfs
+                Err(Errno::EAGAIN) | Err(Errno::EINTR) => continue,
+                Err(_) => return false,
             }
         }
-        Ok(())
     }
 
-    // function similar with fileLabel in go-selinux repo.
-    // fileLabel returns the SELinux label for this path, following symlinks,
-    // or returns an error.
-    pub fn file_label(fpath: &Path) -> Result<String, SELinuxError> {
-        if !fpath.exists() {
-            return Err(SELinuxError::FileLabel(ERR_EMPTY_PATH.to_string()));
+    // find_selinux_fs_mount returns a next selinuxfs mount point found,
+    // if there is one, or an empty string in case of EOF or error.
+    fn find_selinux_fs_mount(s: &str) -> Option<PathBuf> {
```

Comment:
What is `s`?

Reply:
The function's name and the argument's name are vague, so I renamed these.

In go-selinux repo, `s` stands for scanner. But it is not the correct name for this PR.
Therefore I renamed it.